### ISSUE TITLE
simple question

### ISSUE THREAD

**Issue by CodeXmonk** [elias-sundqvist/obsidian-react-components#16]
Hi Elias,

Thank you for your work. It's fantastic. Using it I was able to add a slider (input range) to a note, but there is a simple problem I could not solve:

Here is your example:

```jsx-
<Testcomponent source="Click Me!"/>
```

Is there any way that I could use a frontmatter tag as a source?
I mean:
---
thisIsTheSource: "Click Me!"
---

and after:

```jsx-
<Testcomponent source=`=thisIsTheSource`/> // or something like that?
```
Keep up your nice work.
Sincerelly, Joey

**Comment by elias-sundqvist** [comment]
Yes, it is possible to access the frontmatter. 

If you create a component called "WithFrontMatter"

```jsx
const ctx = useContext(ReactComponentContext);

var ppctx = ctx.markdownPostProcessorContext;
var frontmatter = ppctx.frontmatter;

return props.render(frontmatter)
```
Then you can write notes like this:


````md
---
hello: greetings
---

`jsx- <WithFrontMatter render={({hello})=><h1>The value of hello is "{hello}"</h1>}/>`
````

**Comment by CodeXmonk** [comment]
Warm thanks for your quick answer.
I've just found Obsidian and your plugin opens the door to realization some of my ideas and motivates me to start learning react. I am not a programmer just an old neuropsychologist having some minor experience of javascript ( much earlier: minimal knowledge of Dojo and d3.js but I have not used them for years now) Despite of my ignorance I was able to use you plugin and in preview mode I can have a slider (input range) and with using MetaEdit plugin I could write back the value into the note. Now, thanks to your help, I can give initial value to the slider so next time it shows the value that was stored. 
But first I have to learn... again!-)
Thank you!
József Hornyik (CodeXmonk)
